app/services/coin_market_cap_pro/update_market_pairs_service.rb

```diff
@@ -59,27 +59,47 @@ def update_coin_pairs(coin)
       end
     end
 
+    # Some exchange details have redacted market details (indicated with -1).
     def perform_update_pairs(identifier, data)
       # Grabbing data from snapshot cache
       snapshot = Rails.cache.read("#{identifier}:snapshot")
       base_volume24h = snapshot[:volume24h] unless snapshot.blank?
       has_base_volume24h = base_volume24h.present? && base_volume24h != 0
 
       raw_market_pairs = data.dig("market_pairs")
-      market_pairs = raw_market_pairs.map do |pair|
+      coin_symbol = data.dig("symbol")
+      valid_market_pairs = raw_market_pairs.reject do |pair|
+        pair.dig("quote", currency, "price") == -1 ||
+        pair.dig("quote", currency, "volume_24h") == -1
+      end
+      market_pairs = valid_market_pairs.map do |pair|
         quote = pair.dig("quote", currency)
         volume24h = quote["volume_24h"] || 0
 
+        base_symbol = pair.dig("market_pair_base", "currency_symbol")
+        coin_is_primary = base_symbol == coin_symbol
+
+        volume_percentage = has_base_volume24h ? volume24h / base_volume24h : 0
```

Comment:
Will the result of the division be an `int` or a `float`?

Reply:
I think it's not a big deal since we display integers, but I will force to float anyways